mlflow/agno/autolog.py

```diff
@@ -129,40 +129,82 @@ def _get_span_type(instance) -> str:
     try:
         from agno.agent import Agent
         from agno.models.base import Model
-        from agno.storage.base import Storage
         from agno.team import Team
         from agno.tools.function import FunctionCall
-
     except ImportError:
         return SpanType.UNKNOWN
+
+    storage_types = ()
+    try:
+        # Agno version below 2 uses storage.base
+        from agno.storage.base import Storage
+
+        storage_types = (Storage,)
+    except ImportError:
+        try:
+            # Agno version 2+ uses db.base
+            import agno.db.base as db_base
+
+            candidates = []
+            for attr in ("BaseDb", "AsyncBaseDb"):
+                cls = getattr(db_base, attr, None)
+                if cls is not None:
+                    candidates.append(cls)
+            storage_types = tuple(candidates)
+        except ImportError:
+            pass
+
     if isinstance(instance, (Agent, Team)):
         return SpanType.AGENT
     if isinstance(instance, FunctionCall):
         return SpanType.TOOL
-    if isinstance(instance, Storage):
+    if storage_types and isinstance(instance, storage_types):
         return SpanType.MEMORY
     if isinstance(instance, Model):
         return SpanType.LLM
     return SpanType.UNKNOWN
 
 
+# Agno version >=2 uses Metrics object, but version <2 uses dictionary
 def _parse_usage(result) -> dict[str, int] | None:
     usage = getattr(result, "metrics", None) or getattr(result, "session_metrics", None)
     if not usage:
         return None
 
+    def _get_value(container, key):
+        if isinstance(container, dict):
+            return container.get(key)
+        return getattr(container, key, None)
+
+    def _coerce(value):
```

Comment:
might want to name this for its purpose - it's not general purpose and is only intended to sum or convert scalar values to int type. 
Are guards needed for any reason on validating whether the contents of those collection types are convertible to int type?